buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/integrationtests/shared-configuration.kt

```diff
@@ -95,32 +95,18 @@ fun Project.createTestTask(name: String, executer: String, sourceSet: SourceSet,
 
 
 private
-fun DistributionTest.configureTestSplitIfNecessary(name: String, sourceSet: SourceSet, testType: TestType) {
-    val testSplit = project.stringPropertyOrEmpty("testSplit")
-    if (testSplit.isBlank() || (testType == TestType.CROSSVERSION && name != "crossVersionTest")) {
-        // Cross version tests are splitted by tasks
-        // But if it's crossVersionTest in quickTest, we still split it
-        return
-    }
-
-    val currentSplit = testSplit.split("/")[0].toInt()
-    val numberOfSplits = testSplit.split("/")[1].toInt()
-    val sourceFiles = sourceSet.groovy.files.sortedBy { it.absolutePath }
-
-    if (sourceFiles.size < numberOfSplits || name == "integMultiVersionTest") {
-        // https://github.com/gradle/gradle-private/issues/2740
-        enabled = currentSplit == 1
-        return
-    }
-
-    val buckets = splitIntoBuckets(sourceFiles, numberOfSplits)
-    if (currentSplit == numberOfSplits) {
-        filter.excludePatterns.addAll(buckets.subList(0, buckets.size - 1).flatten().map { it.nameWithoutExtension })
-    } else {
-        filter.includePatterns.addAll(buckets[currentSplit - 1].map { it.nameWithoutExtension })
+fun DistributionTest.configureTestBucketIfNecessary(name: String, sourceSet: SourceSet, testType: TestType) {
+    val includeTestClasses = project.stringPropertyOrEmpty("includeTestClasses")
+    val excludeTestClasses = project.stringPropertyOrEmpty("excludeTestClasses")
```

Comment:
:x: I am afraid this is not enough. This is why CI is failing currently. I think we need separate parameters for the different test types. E.g. `includeTestClassesIntegTest` and `includeTestClassesCrossVersionTest` or something similar. At the moment we apply the filter to all types and then there can be empty lists. And that will make a build fail like this: https://builds.gradle.org/viewLog.html?buildId=29154983&tab=buildResultsDiv&buildTypeId=Gradle_Check_Quick_1_core_2

This is only an issue if we call multiple test types in one Gradle run, as we do for the "quick" phase:

```
    create("quickTest") {
        tasks("test", "integTest", "crossVersionTest")
    }
```